proposals/testing/NNNN-targeted-interoperability-swift-testing-and-xctest.md

```diff
@@ -0,0 +1,292 @@
+
+## Detailed design
+
+### Highlight and support XCTest APIs which are lossy without interop
+
+We propose supporting the following XCTest APIs in Swift Testing:
+
+- Assertions: `XCTAssert*` and [unconditional failure][] `XCTFail`
+- [Expected failures][], such as `XCTExpectFailure`: marking a Swift Testing
+  issue in this way will generate a runtime warning issue.
+- `XCTAttachment`
+- [Issue handling traits][]: we will make our best effort to translate issues
+  from XCTest to Swift Testing. Note that there are certain issue kinds that are
+  new to Swift Testing and not expressible from XCTest.
+
+Note that no changes are proposed for the `XCTSkip` API, because they already
+feature prominently as a test failure to be corrected when thrown in Swift
+Testing.
+
+We also propose highlighting usage of above XCTest APIs in Swift Testing:
+
+- **Report [runtime warning issues][]** for XCTest API usage in Swift Testing.
+  This **applies to assertion successes AND failures**! We want to make sure you
+  can identify opportunities to modernise even if your tests currently pass.
+
+- Opt-in **strict interop mode**, where XCTest API usage will result in
+  `fatalError("Usage of XCTest API in a Swift Testing context is not
+supported")`.
+
+Here are some concrete examples:
+
+| When running a Swift Testing test... | Current         | Proposed                                     | Proposed (strict) |
+| ------------------------------------ | --------------- | -------------------------------------------- | ----------------- |
+| `XCTAssert` failure is a ...         | ‼️ No-op        | ❌ Test Failure and ⚠️ Runtime Warning Issue | 💥 `fatalError`   |
+| `XCTAssert` success is a ...         | No-op           | ⚠️ Runtime Warning Issue                     | 💥 `fatalError`   |
+| `throw XCTSkip` is a ...             | ❌ Test Failure | ❌ Test Failure                              | ❌ Test Failure   |
+
+### Targeted support for Swift Testing APIs with XCTest API equivalents
+
+We propose supporting the following Swift Testing APIs in XCTest:
+
+- `#expect` and `#require`
+  - Includes [exit testing][]
+- `withKnownIssue`: marking an XCTest issue in this way will generate a runtime
+  warning issue. In strict interop mode, this becomes a `fatalError`.
+- Attachments
+- [Test cancellation][] (links to pitch)
+
+We think developers will find utility in using Swift Testing APIs in XCTest. For
+example, you can replace `XCTAssert` with `#expect` in your XCTest tests and
+immediately get the benefits of the newer, more ergonomic API. In the meantime,
+you can incrementally migrate the rest of your test infrastructure to use Swift
+Testing at your own pace.
+
+Present and future Swift Testing APIs will be supported in XCTest if the
+XCTest API _already_ provides similar functionality.
+
+- For example, we support the proposed Swift Testing [test cancellation][]
+  feature in XCTest since it is analogous to `XCTSkip`.
```

Comment:
```suggestion
- For example, if the recently-pitched [test cancellation][] feature in Swift Testing were to be accepted in the future,
  this proposal means that it would be supported in an XCTest test since it's analogous to `XCTSkip`.
```